pkg/diagnostics/tracing.go

```diff
@@ -60,6 +61,34 @@ func startTracingSpanInternal(ctx context.Context, uri, samplingRate string, spa
 	return ctx, span
 }
 
+// GetDefaultSpanContext returns default span context when not provided by the client
+func GetDefaultSpanContext(spec config.TracingSpec) trace.SpanContext {
+	spanContext := trace.SpanContext{}
+
+	gen := tracingConfig.Load().(*traceIDGenerator)
+
+	// Only generating TraceID. SpanID is not generated as there is no span started in the middleware.
+	spanContext.TraceID = gen.NewTraceID()
+
+	rate := diag_utils.GetTraceSamplingRate(spec.SamplingRate)
+
+	// TODO : Continue using ProbabilitySampler till Go SDK starts supporting RateLimiting sampler
+	sampler := trace.ProbabilitySampler(rate)
+	sampled := sampler(trace.SamplingParameters{
+		ParentContext:   trace.SpanContext{},
+		TraceID:         spanContext.TraceID,
+		SpanID:          spanContext.SpanID,
+		HasRemoteParent: false}).Sample
+
+	if sampled {
+		spanContext.TraceOptions |= 1
+	} else {
+		spanContext.TraceOptions &= ^trace.TraceOptions(1)
+	}
```

Comment:
Can we assign sampled to traceOptions directly? spanContext is new one... 
```
spanContext.TraceOptions := sampler(trace.SamplingParameters{ ..
```

Reply:
TraceOptions  is not bool but removed else as your comment.